packages/@ember/-internals/glimmer/lib/component.ts

```diff
@@ -29,15 +29,14 @@ export const BOUNDS = symbol('BOUNDS');
 */
 
 /**
-  A `Component` is a view that is completely
-  isolated. Properties accessed in its templates go
-  to the view object and actions are targeted at
-  the view object. There is no access to the
-  surrounding context or outer controller; all
-  contextual information must be passed in.
-
-  The easiest way to create a `Component` is via
-  a template. If you name a template
+  A component is an isolated piece of UI, represented by a template and an
+  optional class. When a component has a class, its template's `this` value
```

Comment:
I find the "When a component..." sentence confusing, though accurate. It's ok as-is (shouldn't block a merge)

Reply:
In general, I had trouble documenting "components" in `Ember.Component` and think we need a better general strategy going forward. This was the best thing I could come up with for now, but agree we should focus on how to document component APIs in the API docs ASAP.